File: aspnet-core/src/ERP.Healthcare.Application.Contracts/Doctors/DoctorDto.cs
using System.Collections.Generic;
using Volo.Abp.Application.Dtos;
using Volo.Abp.Data;

namespace ERP.Healthcare.Doctors
{
    public class DoctorDto : EntityDto<int> 
    {
        public string Name { get; set; }

        public string Description { get; set; }

        public GenderTypeEnum Gender { get; set; }

        public int AddressId { get; set; }

        public int DoctorSpecialtyId { get; set; }

        public int DoctorTitleId { get; set; }

        public AddressDto AddressDto { get; set; }

        public DoctorSpecialtyDto DoctorSpecialtyDto { get; set; }

        public DoctorTitleDto DoctorTitleDto { get; set; }
    }
}

File: angular/src/app/doctor/models/doctor-dto.ts

import { DoctorTitleDto } from './doctor-title-dto';
import { DoctorSpecialtyDto } from './doctor-specialty-dto';
import { AddressDto } from './address-dto';

import { EntityDto } from '@abp/ng.core';

export class DoctorDto extends EntityDto<number> {
  name: string;
  description: string;
  gender: any;
  addressId: number;
  doctorSpecialtyId: number;
  doctorTitleId: number;
  addressDto: any;
  doctorSpecialtyDto: any;
  doctorTitleDto: any;
  id: number;

  constructor(initialValues: Partial<DoctorDto> = {}) {
    super(initialValues);
  }
}
